src/runtime/import.cpp

```diff
@@ -140,19 +147,119 @@ SearchResult findModule(const std::string& name, const std::string& full_name, B
     return SearchResult("", SearchResult::SEARCH_ERROR);
 }
 
+/* Return the package that an import is being performed in.  If globals comes
+   from the module foo.bar.bat (not itself a package), this returns the
+   sys.modules entry for foo.bar.  If globals is from a package's __init__.py,
+   the package's entry in sys.modules is returned, as a borrowed reference.
+
+   The *name* of the returned package is returned in buf.
+
+   If globals doesn't come from a package or a module in a package, or a
+   corresponding entry is not found in sys.modules, Py_None is returned.
+*/
+static Box* get_parent(Box* globals, int level, std::string& buf) {
```

Comment:
What do you think about naming this `getParent()`?  It looks like this was mostly copied from CPython, but if it's a Pyston-style function now (throws Pyston exceptions), using the Pyston-style naming could help make that clear to callers.

I wish we had a better way of annotating this distinction :/